### ISSUE TITLE
Direct Debit Information 'profile' help text

### ISSUE THREAD

**Issue by magnolia61** [Project60/sepa_dd#221]
In order to help the parents who hopefully will be filling in lots of donation and event registration forms using civisepa, we use Pre-field help and Pre-form Help a lot in our forms.

Is there a way we can add pre-field or pre-form help information to the "Direct Debit Information" section?

**Comment by systopia** [comment]
@magnolia61 Could you be more precise?

In general, if the system as-is doesn't support pre/post form help it is not our intention to develop this within the scope of this extension, that would be a general feature request IMHO. 

As a workaround, however, you could inject such texts into the page using the appropriate hooks in your SEPA customization extension.

**Comment by magnolia61** [comment]
I made a mockup to help visualize what I mean. Many people in the Netherlands still do not know their full IBAN account number. And absolutely nobody knows what a BIC number is. 
Also some friendly (and necessary) text to let people know they are giving a mandate would be nice and helps prevent all kind of questions being asked by mail/phone/etc. Especially for non-profits working with volunteers the forms should be self-explanitory. The purpose of civisepa I guess is to be widely and happily used in europe :-)

I think we even would be ok if it is going to be only 1 configurable field per creditor (so only pre-form help text) and be configurable in the creditor settings. For sophisticated and specialized needs I see a great use for customization extensions, but basic stuff like help text (and even hiding BIC of startdate) I think the creditor setting provides the easiest place for non-developers to configure  civisepa. I think this will benefit the acceptance and easy of use for semi-tech people (like me) a lot!

![civicrm_sepadd help text new](https://cloud.githubusercontent.com/assets/2195908/5595566/db5d35f2-9276-11e4-9e25-5aa200a412b9.png)

**Comment by systopia** [comment]
As you know, the BIC is still mandatory, see https://github.com/Project60/sepa_dd/issues/209. However, if you install the ["Little BIC Extension"](https://github.com/Project60/org.project60.bic) and auto-download the Dutch BIC numbers, it should be filled automatically when the user enters his/her IBAN. You could even add some JavaScript to hide the BIC field, and only show it in case the BIC could not be determined automatically.

With respect to the extra fields depicted in your sketch, I still think this should be injected by your customisation extension. Since you should have one anyway, I don't see the harm in that - shouldn't be a lot of work. This approach allows us to keep the main SEPA extension more generic and flexible.

**Comment by magnolia61** [comment]
i downloaded & installed the little BIC extension. Downloading the banklist goes ok. Should it automagically fill the BIC after a press TAB after entering the IBAN? Because it does not. I didn't know this functionality should already be functional. Would love to have it working though.

On the help texts. I understand you hesitation. But at least there should be some info on creating a mandate because now people create one unknowingly. Maybe it's possible you just hardcode a non-configurable message on mandate creation in there. So at least people know what they are doing?

**Comment by systopia** [comment]
> Should it automagically fill the BIC after a press TAB after entering the IBAN? Because it does not.

It should, as long as you're using the ```dev_pp``` branch (which you said you do). If you do, but it still doesn't work, there might be something wrong with the NL parser. In that case please raise a ticket in the other project. See also: https://github.com/Project60/sepa_dd/issues/211.

With respect to the help texts: I still believe this should be up to donation page configuration, just like all the other pre/post help texts. If we just put something in, other people *will* complain, that didn't have that message so far. I'm not quite sure why you are uncomfortable with putting it in the customization extension. Do you know what I mean by that?

**Comment by magnolia61** [comment]
Actually I would love to learn more on the customization extension. If you think it's the way to go for the help text. I am much more able to modify and re-use stuff than code on my own, so I feel at least a quite scared to get my head into it... :-)

**Comment by antrik** [comment]
On Fri, Jan 02, 2015 at 07:46:10AM -0800, SYSTOPIA wrote:
> With respect to the help texts: I still believe this should be up to donation page configuration, just like all the other pre/post help texts.

The Mandate explanations are specific to the SEPA PP, and should only show up when this processor is selected. So on Donation Pages with a choice of PPs, they can not go in the Donation Page configuration.

Note that the SEPA Rulebook prescribes various mandatory items for a SEPA Mandate -- including a certain explanation text. The wording of this text is actually prescribed in the Rulebook -- so there is no reason for omitting it from the standard extension...

(A customisation hook is probably still useful though for adding translated versions of the text.)

**Comment by tttp** [comment]
@olaf,

I think you are confusing the mandate and the form to prepare the mandate.

Is the rulebook mentioning anything about the form?

**Comment by antrik** [comment]
@tttp No, I'm not confusing it -- it's just a very different situation than what you have in mind :-)

The way you are handling SEPA, sending out actual paper mandates to sign, the SEPA fields on the Contribution Page are indeed only for getting information to create the actual mandate. For us in Germany (and apparently also Netherlands), where mandates filled online are considered kinda valid, and thus we omit the paper mandate, it's an entirely different story: the form on the Contribution Page indeed *is* the mandate, and thus I'm pretty sure it should fulfill the formal requirements...

**Comment by tttp** [comment]
On 7 January 2015 at 10:16, Olaf Buddenhagen <notifications@github.com>
wrote:

> @tttp No, I'm not confusing it -- it's just a very different situation
> than what you have in mind :-)
>
> The way you are handling SEPA, sending out actual paper mandates to sign,
> the SEPA fields on the Contribution Page are indeed only for getting
> information to create the actual mandate. For us in Germany (and apparently
> also Netherlands), where mandates filled online are considered kinda valid,
> and thus we omit the paper mandate, it's an entirely different story: the
> form on the Contribution Page indeed *is* the mandate, and thus I'm pretty
> sure it should fulfill the formal requirements...
>
>
If anything, it's the confirmation page that is the closest to the mandate,
for instance because the form doesn't have yet an id until it's processed.

**Comment by antrik** [comment]
> If anything, it's the confirmation page that is the closest to the mandate,

I guess that's also a possible way to handle it... Note though that confirmation pages are optional -- I for my part would always disable them :-)

> for instance because the form doesn't have yet an id until it's processed.

You don't need the Mandate Reference at the time of signing the mandate. It can be supplied later in the Pre-Notification.

**Comment by Detsieber** [comment]
No, the confirmation page is not the right place for the mandate info
text. The mandate text *has to be* on the same page, where the mandate
info is entered!
(I think, we all would not like to sign a mandate, and afterwards get
the information what exactly we signed before...)

Is it really so hard to put that text onto the contribution page, in a
way they only show up when the Payment Processor SEPA DD is selected?


Cheers,
Detlev

Am 07.01.2015 um 18:52 schrieb Olaf Buddenhagen:

**Comment by tttp** [comment]
Not arguing on the need, it'd be better, but olaf reasoning doesn't make
much sense to bring contractual reasons on a practice that isn't part of
the sepa contract ;)

PR/funding welcome and all that ;)

**Comment by magnolia61** [comment]
I think adding a configurable field to the creditor configuration would be by far the easiest and would fill everyones needs. If left blank nothing happens. If filled it would give the possible member/participant/donor some more info on what it means to choose this payment processor. Who would be able to code a PR for this?

**Comment by antrik** [comment]
On Thu, Jan 08, 2015 at 01:48:48AM -0800, xavier dutoit wrote:

> Not arguing on the need, it'd be better, but olaf reasoning doesn't
> make much sense to bring contractual reasons on a practice that isn't
> part of the sepa contract ;)

Not sure what you are saying there -- but again, the way German
organisations handle DD, entering the account data on the contribution
page *is* effectively signing the mandate, and thus following the formal
requirements for a mandate is a must to give it any kind of validity.

> PR/funding welcome and all that ;)

Now that's quite a different position than the one previously expressed
(by systopia), which amounted to "this shouldn't become part of the
standard version at all" :-)

**Comment by systopia** [comment]
> Now that's quite a different position than the one previously expressed (by systopia), which amounted to "this shouldn't become part of the standard version at all" :-)

I don't think it's sensible to put it into the standard version, that's true. But if you (people) come up with a good concept and a PR/funding, of course I wouldn't block it.
